interfaces/matlab_experimental/1D/Sim1D.m

```diff
@@ -0,0 +1,537 @@
+
+        end
+
+        %% Sim1D Class Destructor
+
+        function delete(s)
+            % Delete the :mat:class:`Sim1D` object.
+
+            ctFunc('sim1D_del', s.stID);
+        end
+
+        %% Sim1D Utility Methods
+
+        function display(s, fname)
+            % Show all domains.
+
+            if nargin == 1
+                fname = '-';
+            end
+
+            ctFunc('sim1D_showSolution', s.stID, fname);
+        end
+
+        function plotSolution(s, domain, component)
+            % Plot a specified solution component. ::
+            %
+            %     >> s.plotSolution(domain, component)
+            %
+            % :param s:
+            %    Instance of class :mat:class:`Sim1D`.
+            % :param domain:
+            %    Name of domain from which the component should be
+            %    retrieved.
+            % :param component:
+            %    Name of the component to be plotted.
+
+            n = s.stackIndex(domain);
+            d = s.domains{n};
+            z = d.gridPoints;
+            x = s.solution(domain, component);
+            plot(z, x);
+            xlabel('z (m)');
+            ylabel(component);
+        end
+
+        function restore(s, fname, id)
+            % Restore a previously-saved solution. ::
+            %
+            %     >> s.restore(fname, id)
+            %
+            % This method can be used to provide an initial guess for the solution.
+            %
+            % See also: :mat:class:`save`
+            %
+            % :param s:
+            %     Instance of class :mat:class:`Sim1D`.
+            % :param fname:
+            %     File name of an YAML or HDF file containing solution information.
+            % :param id:
+            %     ID of the element that should be restored.
+
+            ctFunc('sim1D_restore', s.stID, fname, id)
+        end
+
+        function saveSoln(s, fname, id, desc)
+            % Save a solution to a file. ::
+            %
+            %     >> s.saveSoln(fname, id, desc)
+            %
+            % The output file is in a format that can be used by :mat:class:`restore`
+            %
+            % :param s:
+            %     Instance of class :mat:class:`Sim1D`.
+            % :param fname:
+            %     File name where YAML or HDL file should be written.
+            % :param id:
+            %     ID to be assigned to the file element when it is written.
+            % :param desc:
+            %     Description to be written to the output file.
+
+            if nargin == 1
+                fname = 'soln.yaml';
+                id = 'solution';
+                desc = '--';
+            elseif nargin == 2
+                id = 'solution';
+                desc = '--';
+            elseif nargin == 3
+                desc = '--';
+            end
+
+            ctFunc('sim1D_save', s.stID, fname, id, desc);
+        end
+
+        function x = solution(s, domain, component)
```

Comment:
Needs to be renamed, as `solution` is reserved for `Solution` to avoid confusion.